cmake/Pcsx2Utils.cmake

```diff
@@ -70,6 +70,11 @@ function(check_compiler_version version_warn version_err)
             endif()
         endif()
 
+        if(GCC_VERSION VERSION_EQUAL "7.0" OR GCC_VERSION VERSION_EQUAL "7.1")
+            message(FATAL_ERROR "GCC 7.0/7.1  generates invalid code => https://gcc.gnu.org/bugzilla/show_bug.cgi?id=80799\n"
+            "You can either backport the fix or swith to another version of GCC.")
```

Comment:
I suggest this be a warning and then it disables the known bad optimizations as the current workaround already does. If a distro does patch it, they probably can also patch pcx2 to remove the workaround until it is no longer needed.